perception_dataset/deepen/segmentation/painting2d.py

```diff
@@ -0,0 +1,205 @@
+from __future__ import annotations
+
+import base64
+from dataclasses import dataclass
+from glob import glob
+import json
+import os.path as osp
+import shutil
+from typing import Any, Dict, List, Tuple
+
+from PIL import Image
+import numpy as np
+from numpy.typing import NDArray
+import pycocotools.mask as cocomask
+import skimage
+
+from perception_dataset.deepen.deepen_annotation import DeepenAnnotation, LabelType
+
+__all__ = ["DeepenSegmentationPainting2D"]
+
+
+@dataclass
+class DeepenSegmentationPainting2D(DeepenAnnotation):
+    """A class to handle 2D segmentation annotation annotated by painting."""
+
+    @classmethod
+    def from_file(
+        cls,
+        ann_file: str,
+        data_root: str,
+        camera2index: Dict[str, int],
+        dataset_corresponding: Dict[str, str],
+        *,
+        as_dict: bool = True,
+    ) -> List[DeepenSegmentationPainting2D]:
+        """Return a list of `DeepenSegmentationPainting2D`s from files.
+
+        Args:
+            ann_file (str): Annotation files path compressed as `.zip`.
+            data_root (str): Root directory of the T4 dataset.
+            camera2index (Dict[str, int]): Name mapping from camera name to camera index.
+            dataset_corresponding (Dict[str, str]): Key-value mapping of T4 dataset name and Deepen ID.
+            as_dict (bool, optional): Whether to output objects as dict or its instance.
+                Defaults to True.
+
+        Returns:
+            List[DeepenSegmentationPainting2D]: List of converted `DeepenSegmentationPainting2D`s.
+        """
+        output: List[DeepenSegmentationPainting2D] = []
+        for t4_dataset_name, dataset_id in dataset_corresponding.items():
+
+            # Parse annotation ZIP file
+            annotations = _parse_annotation(ann_file, camera2index)
+
+            for ann_info, id2category in annotations:
+                # Get image size
+                image = Image.open(osp.join(data_root, t4_dataset_name, ann_info.image))
+                width, height = image.size
+
+                # Load mask from .npy
+                mask: NDArray = np.load(ann_info.mask)
+                mask = mask.reshape((height, width))
+
+                # Get sensor ID
+                sensor_id = f"sensor{camera2index[ann_info.camera_name]}"
```

Comment:
```suggestion
                sensor_id = f"sensor{camera2index[ann_info.camera_name] - 1}"
```
To keep consistency between deepen and t4, we have to subtract `1` here.

- Deepen: sensor1, sensor2,,,,
- t4: camera0, camera1,,,,